app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -97,6 +101,8 @@ fun MapScreen(
   var pointAnnotationManager by remember { mutableStateOf<PointAnnotationManager?>(null) }
   val selectedParking by parkingViewModel.selectedParking.collectAsState()
 
+  val context = LocalContext.current as Activity
```

Comment:
I do not realy understand why you pass down `activity` through the whole app when you can just have it with `LocalContext.current`, does this not work?

Reply:
It's a mistake on my part, it is not used anywhere so forgot to remove it

Reply:
No, what I meant is can you not use that `context` variable as initializer to the `PermissionHandler`?
Instead of adding an extra argument everywhere

Reply:
I need to get the Context from the Main which is not the same as the one in the Map. Since the permissions are asked in the context of the Main I need to bring it to Map the same way i bring the permission Manager so that i can correctly check if location permissions were given.

Reply:
Ok, i tried and it seemed you were right so i cleaned the code a little, feel free to test it again.

Reply:
I agree w/ @matttsss that removing the extra variables make the code cleaner. I've re-tested to ensure that everything still works and it does!